Userland/Games/Flood/main.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "BoardWidget.h"
+#include "SettingsDialog.h"
+#include <AK/URL.h>
+#include <LibConfig/Client.h>
+#include <LibCore/ConfigFile.h>
+#include <LibCore/System.h>
+#include <LibDesktop/Launcher.h>
+#include <LibGUI/Action.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Icon.h>
+#include <LibGUI/MessageBox.h>
+#include <LibGUI/Statusbar.h>
+#include <LibGUI/Window.h>
+#include <LibGfx/Painter.h>
+#include <LibMain/Main.h>
+
+// Adapted from Libraries/LibVT/TerminalWidget.cpp::TerminalWidget::set_color_scheme.
+static ErrorOr<Vector<Color>> get_color_scheme_from_string(StringView name)
+{
+    if (name.contains('/')) {
+        return Error::from_string_literal("Shenanigans! Color scheme names can't contain slashes.");
+    }
+
+    constexpr StringView color_names[] = {
+        "Black"sv,
+        "Red"sv,
+        "Green"sv,
+        "Yellow"sv,
+        "Blue"sv,
+        "Magenta"sv,
+        "Cyan"sv,
+        "White"sv
+    };
+
+    auto const path = String::formatted("/res/terminal-colors/{}.ini", name);
+    auto color_config_or_error = Core::ConfigFile::open(path);
+    if (color_config_or_error.is_error()) {
+        return Error::from_string_view(String::formatted("Unable to read color scheme file '{}': {}", path, color_config_or_error.error()));
+    }
+    auto const color_config = color_config_or_error.release_value();
+    Vector<Color> colors;
+
+    for (u8 color_idx = 0; color_idx < 8; ++color_idx) {
+        auto const rgb = Gfx::Color::from_string(color_config->read_entry("Bright", color_names[color_idx]));
+        if (rgb.has_value())
+            colors.append(Color::from_argb(rgb.value().value()));
+    }
+
+    auto const default_background = Gfx::Color::from_string(color_config->read_entry("Primary", "Background"));
+    if (default_background.has_value())
+        colors.append(default_background.value());
+    else
+        colors.append(Color::DarkGray);
+
+    return colors;
+}
+
+static int get_number_of_moves_from_ai(Board const& board)
```

Comment:
It looks like the approach taken by this AI is not actually optimal, just a good first attempt; it might be better to get more "touching points" to tiles first before flooding a lot of tiles as that can be done in `color_count` moves when all tiles are touched. So please leave a comment stating something like this, so that we might improve it in the future, especially when considering a user-facing auto-solver.